src/_nebari/stages/infrastructure/__init__.py

```diff
@@ -743,15 +750,27 @@ def tf_objects(self) -> List[Dict]:
             ]
         elif self.config.provider == schema.ProviderEnum.azure:
             return [
+                *resources,
+                NebariOpentofuRequiredProvider("azurerm", self.config),
                 NebariTerraformState(self.name, self.config),
             ]
         elif self.config.provider == schema.ProviderEnum.aws:
             return [
+                *resources,
+                NebariOpentofuRequiredProvider("aws", self.config),
                 opentofu.Provider("aws", region=self.config.amazon_web_services.region),
                 NebariTerraformState(self.name, self.config),
             ]
+        elif self.config.provider == schema.ProviderEnum.local:
+            return [
+                *resources,
+                NebariOpentofuRequiredProvider("kind", self.config),
```

Comment:
I think originally, I had in mind a `*RequiredProviders` class for accommodating all providers for each stage in a single sweep, but this also does the job.